app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java

```diff
@@ -195,17 +193,28 @@ private void setAchievements() {
                                 if (response != null) {
                                     setUploadCount(Achievements.from(response));
                                 } else {
-                                    onError();
+                                    showSnackBarWithRetry();
                                 }
                             },
                             t -> {
                                 Timber.e(t, "Fetching achievements statistics failed");
-                                onError();
+                                showSnackBarWithRetry();
                             }
                     ));
         }
     }
 
+    private void showSnackBarWithRetry() {
+        progressBar.setVisibility(View.GONE);
+        if (onClickListener == null) {
```

Comment:
just curious why do you need to maintain `View.OnClickListener` and do a null check. The function below will set the click listener and not add one. So theres no need to take care of singleton.

Reply:
Yeah, have replaced the click listener with Java 8 Lambda function.